acceleratorparse.js

```diff
@@ -1,2375 +1,2382 @@
 /**
  * Provides replacement for Gtk.accelerator_parse.
- * @param {String} keystr
  */
-function accelerator_parse(keystr) {
-    let mask = accelerator_mask(keystr);
-    const mods = accelerator_mods(keystr);
-
-    // remove mods from keystr
-    let key = keystr;
-    mods.forEach(m => {
-        key = key.replace(m, '');
-    });
-    key = key.trim();
-
-    // now lookup key in map
-    let ok = true;
-    const mapped = keycodeMap.get(key);
-    if (mapped) {
-        key = mapped;
-    }
-    else {
-        ok = false;
-        key = 0;
-        mask = 0;
+export class AcceleratorParse {
```

Comment:
Is there a point to having this as a class? You are still basically only calling `accelerator_parse` but now you have to pass around an object.

Reply:
Thanks @Lythenas, yes, there is specific reason for this.  So, it was originally a module - issue with modules with top-level fields, issue with top-level fields is that the populate/pollute global scope in Gnome shell.  Which is okay, but they must be explicitly cleaned on on `disable`.

However, `pref.js` is not running on the gnome-shell process (runs in own/separate process)... and has no `disable` - but it still can populate global Gnome scope.  So, we were using a hack to clean this up `prefs.js` (essentially detected `close-request` events).  Gnome team asked us to change this, especially for Gnome 45.

Hence, it's now an object (inside a `prefs.js` class) that doesn't pollute global scope - and hence doesn't require explicit cleaning up.

Note too, that there's only one `accelerator_parse` object in PaperWM (gnome-shell process) and another one when opening up `prefs` (note there's no comms between PaperWM extension and prefs - although we can share modules or detected changes via `gsettings`).

Anyways, this was recommended and requested to change to object by Gnome team:
![image](https://github.com/paperwm/PaperWM/assets/30424662/192789e1-2b18-4068-a226-c5353697e60c)